src/PowerShellEditorServices/Services/Symbols/Vistors/AstOperations.cs

```diff
@@ -153,42 +154,21 @@ public static SymbolReference FindCommandAtPosition(Ast scriptAst, int lineNumbe
         /// </summary>
         /// <param name="scriptAst">The abstract syntax tree of the given script</param>
         /// <param name="symbolReference">The symbol that we are looking for referneces of</param>
-        /// <param name="CmdletToAliasDictionary">Dictionary maping cmdlets to aliases for finding alias references</param>
-        /// <param name="AliasToCmdletDictionary">Dictionary maping aliases to cmdlets for finding alias references</param>
+        /// <param name="cmdletToAliasDictionary">Dictionary maping cmdlets to aliases for finding alias references</param>
+        /// <param name="aliasToCmdletDictionary">Dictionary maping aliases to cmdlets for finding alias references</param>
         /// <returns></returns>
         public static IEnumerable<SymbolReference> FindReferencesOfSymbol(
             Ast scriptAst,
             SymbolReference symbolReference,
-            Dictionary<String, List<String>> CmdletToAliasDictionary,
-            Dictionary<String, String> AliasToCmdletDictionary)
+            ConcurrentDictionary<string, List<string>> cmdletToAliasDictionary = default,
+            ConcurrentDictionary<string, string> aliasToCmdletDictionary = default)
```

Comment:
I'm curious about the change from `Dictionary` to `ConccurrentDictionary`. Can you please briefly describe the scenarios where concurrent updates to the dictionary is needed?

Reply:
Really just that there is no guarantee that `GetAliasesAsync` will be called with a lock around it, so concurrent requests to resolve references may concurrently ask for aliases to be resolved, and so concurrently access the original `ConcurrentDictionary` caches in `CommandHelpers.cs`. Now, `AstOperations.FindReferences` is called with a newly constructed instance each time, so _it_ should be thread safe since it's passed copies of the cache dictionaries, but it was easiest to just make the types match. I suppose I should instead receive an `IDictionary<K,V>` here.

Reply:
Fixed!